src/main/java/com/controller/QuoteController.java

```diff
@@ -44,18 +44,31 @@ public ResponseEntity createQuote(@RequestBody Quote quote) {
      */
     @GetMapping(path = "/quotes", produces = "application/json")
     public ResponseEntity<List<Quote>> getQuotes(@RequestParam(required = false) String author,
+                                                 @RequestParam(required = false) String source,
                                                  @RequestParam(required = false) Integer rating) {
         List<Quote> allQuotes = null;
-        if (!isEmpty(author) && !isEmpty(rating)) {
+        if (!isEmpty(author) && !isEmpty(rating) && !isEmpty(source)) {
```

Comment:
you may try to generate the method name by the available fields and then invoke it by reflection

Reply:
..but in new merge request

Reply:
will be adressed